jlm/llvm/opt/alias-analyses/Steensgaard.cpp

```diff
@@ -1733,6 +1754,37 @@ Steensgaard::Analyze(
   return pointsToGraph;
 }
 
+void
+Steensgaard::PropagatePointsToFlags()
+{
+  bool pointsToFlagsChanged;
+  do
+  {
+    pointsToFlagsChanged = false;
+
+    for (auto & set : *LocationSet_)
+    {
+      auto location = set.value();
+
+      // Nothing needs to be done if this set does not point to another set
+      if (!location->GetPointsTo())
+      {
+        continue;
+      }
+      auto & pointsToLocation = LocationSet_->GetRootLocation(*location->GetPointsTo());
+
+      auto locationFlags = location->GetPointsToFlags();
+      auto pointsToLocationFlags = pointsToLocation.GetPointsToFlags();
+      auto combinedFlags = locationFlags | pointsToLocationFlags;
+      if (pointsToLocationFlags != combinedFlags)
+      {
+        pointsToLocation.SetPointsToFlags(combinedFlags);
+        pointsToFlagsChanged = true;
+      }
+    }
+  } while (pointsToFlagsChanged);
```

Comment:
This algorithm is a simple fix-point computation. It behaves quadratic in the number of disjoint location sets. I can foresee that its runtime might become a problem in the future. However, I currently care more about correctness than performance. Once the data shows that it is a problem, I will tackle it. Thus, please ignore its performance aspect right now.